docs/source/tutorial/interactions.rst

```diff
@@ -0,0 +1,485 @@
+    \rangle
+
+for quaternions $\mathbf{h}, \mathbf{r}, \mathbf{t} \in \mathbf{H}^{d}$, and Hamilton product $\otimes$.
+
+HolE
+~~~~
+
+:class:`~pykeen.nn.modules.HolEInteraction` is given by
+
+.. math::
+
+    \langle \mathbf{r}, \mathbf{h} \star \mathbf{t}\rangle
+
+where $\star: \mathbb{R}^d \times \mathbb{R}^d \rightarrow \mathbb{R}^d$ denotes the circular correlation:
+
+.. math::
+
+    [\mathbf{a} \star \mathbf{b}]_i = \sum_{k=0}^{d-1} \mathbf{a}_{k} * \mathbf{b}_{(i+k)\ \mod \ d}
+
+AutoSF
+~~~~~~
+
+:class:`~pykeen.nn.modules.AutoSFInteraction` is an attempt to parametrize *block-based* semantic matching interaction
+functions to enable automated search across those. Its interaction is given as
+
+.. math::
+
+    \sum_{(i_h, i_r, i_t, s) \in \mathcal{C}} s \cdot \langle h[i_h], r[i_r], t[i_t] \rangle
+
+where $\mathcal{C}$ defines the block interactions, and $h, r, t$ are lists of blocks.
+
+Neural Interactions
+-------------------
+
+All other interaction functions are usually called *neural*. They share that they usually have a multi-layer
+architecture (usually two) and employ non-linearities. Many of them also introduce customized hidden layers such as
+interpreting concatenated embedding vectors as image, pairs of embedding vectors as normal distributions, or semantic
+matching inspired sums of linear products.
+
+Moreover, some choose a form that can be decomposed into
+
+.. math::
+
+    f(\mathbf{h}, \mathbf{r}, \mathbf{t}) = f_o(f_i(\mathbf{h}, \mathbf{r}), \mathbf{t})
+
+with an expensive $f_i$ and a cheap $f_o$. Such form allows efficient scoring of many tails for a given head-relation
+combination, and can becombined with inverse relation modelling for an overall efficient training and inference
+architecture.
+
+ConvE
+~~~~~
+
+:class:`~pykeen.nn.modules.ConvEInteraction` uses an interaction of the form
+
+.. math::
+
+    \langle g(\mathbf{h}, \mathbf{r}), \mathbf{t} \rangle + t_b
+
+for $\mathbf{h}, \mathbf{r}, \mathbf{t} \in \mathbb{R}^d$ are the head entity, relation, and tail entity representation,
+and $t_b \in \mathbb{R}$ is an entity bias. $g$ is a CNN-based encoder, which first operates on a 2D-reshaped "image"
+and then flattens the output for a second linear layer. Dropout and batch normalization is utilized, too.
+
+ConvKB
+~~~~~~
+
+:class:`~pykeen.nn.modules.ConvKBInteraction` concatenates $\mathbf{h}, \mathbf{r}, \mathbf{t} \in \mathbb{R}^d$ to a $3
+\times d$ "image" and applies a $3 \times 1$ convolution. The output is flattened and a linear layer predicts the score.
+
+CrossE
+~~~~~~
+
+:class:`~pykeen.nn.modules.CrossEInteraction` uses
+
+.. math::
+
+    \langle g(\mathbf{h}, \mathbf{r}), \mathbf{t} \rangle
+
+where
+
+.. math::
+
+    g(\mathbf{h}, \mathbf{r}) = \sigma(
+        \mathbf{c}_r \odot \mathbf{h}
+        + \mathbf{c}_r \odot \mathbf{h} \odot \mathbf{r}
+        + \mathbf{b}
+    )
+
+with an activation function $\sigma$. Moreover, dropout is applied to the output of $g$.
```

Comment:
explicitly say what odot operator is

Reply:
https://github.com/pykeen/pykeen/pull/1484/commits/2db0f6e4ccacdf049acacc3115f64444c846c49e